### PR TITLE
[BOP-1195] Initial docs for kubelet configuration

### PR DESCRIPTION
**Pull request by nsteph**
This PR creates some initial docs for configuring the various kube components, with special attention paid to kubelet and kubelet custom profiles.

### CODE REVIEW

File: content/docs/operations/kubernetes/kubelet-custom-profiles.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+weight: 2
+---
+
+Kubelet may be configured on a per-node basis using custom profiles. A Kubelet custom profile consists of a profile name and a set of values. The profile name is used to identify the profile and to target it to specific nodes in the cluster. The values are merged into the final Kubelet configuration applied to targeted node.
+
+## Creating a custom profile
+
+Custom profiles may be specified in the `kubelet.customProfiles` section of the MKE config. Every profile must have a unique name. Values may refer to fields in the Kubelet configuration file. For all possible values, see: [kubelet-config-file](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/).
+
+Example: this configuration creates a custom profile named `hardworker` that specifies thresholds for garbage collection of images and eviction.
```
Comment (KoryKessel-Mirantis) [inline content/docs/operations/kubernetes/kubelet-custom-profiles.md:12]:
```suggestion
The following example configuration creates a custom profile named `hardworker` that specifies thresholds for the garbage collection of images and eviction.
```

File: content/docs/operations/kubernetes/kubelet-custom-profiles.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          evictionHard:
+            imagefs.available: 15%
+            memory.available: 100Mi
+            nodefs.available: 10%
+            nodefs.inodesFree: 5%
+```
+
+## Applying a custom profile to a node
+
+Hosts may be assigned a custom profile using the `Hosts` section of the MKE config. The profile name is an install-time argument for the host.
```
Comment (KoryKessel-Mirantis) [inline content/docs/operations/kubernetes/kubelet-custom-profiles.md:31]:
```suggestion
Hosts can be assigned a custom profile through the `hosts` section of the MKE configuration file, whereas the profile name is an install time argument for the host.
```

File: content/docs/operations/kubernetes/kubelet-custom-profiles.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+Example: this configuration applies the `hardworker` custom profile to the `localhost` node.
```
Comment (KoryKessel-Mirantis) [inline content/docs/operations/kubernetes/kubelet-custom-profiles.md:33]:
```suggestion
The following example configuration applies the `hardworker` custom profile to the `localhost` node.
```

File: content/docs/operations/kubernetes/kubelet-custom-profiles.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      keyPath: ~/.ssh/id_rsa
+      port: 22
+      user: root
+      installFlags:
+        - --profile=hardworker
+```
+
+## Precedence of Kubelet configuration
+
+The Kubelet configuration of each node is created by merging a number of different configuration sources. For MKE, the order is as follows:
```
Comment (KoryKessel-Mirantis) [inline content/docs/operations/kubernetes/kubelet-custom-profiles.md:49]:
```suggestion
The Kubelet configuration of each node is created by merging several different configuration sources. For MKE 4, the order is as follows:
```

File: content/docs/operations/kubernetes/kubelet-custom-profiles.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+- Structured configuration values specified in the `kubelet` section of the MKE config (lowest precedence)
+- Custom profile values specified in `kublelet.customProfiles`
+- Runtime flags specified in `kubelet.extraArgs` (highest precedence)
```
Comment (KoryKessel-Mirantis) [inline content/docs/operations/kubernetes/kubelet-custom-profiles.md:53]:
```suggestion
1. Structured configuration values specified in the `kubelet` section of the MKE configuration, which is the lowest precedence.
2. Custom profile values specified in `kublelet.customProfiles`.
3. Runtime flags specified in `kubelet.extraArgs`, which is the highest precedence.
```

File: content/docs/operations/kubernetes/kubelet-custom-profiles.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+For a more information about the precedence of Kubelet configuration values, see [kubelet-configuration-merging-order](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/#kubelet-configuration-merging-order).
```
Comment (KoryKessel-Mirantis) [inline content/docs/operations/kubernetes/kubelet-custom-profiles.md:55]:
```suggestion
For more information on Kubelet configuration value precedence, refer to the official Kubernetes documentation [Kubelet configuration merging order](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/#kubelet-configuration-merging-order).
```

**Review by KoryKessel-Mirantis — CHANGES_REQUESTED** [review]

File: content/docs/operations/kubernetes/_index.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+      cpu: 50m
+      ephemeral-storage: 500Mi
+      memory: 300Mi
+    managerKubeReserved:
+      cpu: 250m
+      ephemeral-storage: 4Gi
+      memory: 2Gi
+```
+
+Advanced: the kubelet may be further configured using the `extraArgs` field. This map creates runtime flags that are passed directly to the kubelet process. Flags configured in this way are applied with the highest precedence.
```
[the remark this answers is not in the document]
Comment (nsteph) [reply content/docs/operations/kubernetes/_index.md]: I tried to rewrite these sections for clarity. Does this read better?

"You can further configure a kubelet using the `extraArgs` field to define flags. This field accepts a list of key-value pairs, which are passed directly to the kubelet process at runtime."

**Review by KoryKessel-Mirantis — APPROVED** [review]

**Pull request merged**